### PR TITLE
Anonymous login

### PR DESCRIPTION
**Pull request by waxlamp**
This adds the "anonymous login" functionality. Essentially:
- Any time you log out of the app explicitly, the system implicitly logs in a special user, but makes the UI look like no one is logged in. This lets people use the application without first having to create an account.
- The username can be supplied via an environment variable `OSUMO_ANON_USER` at Girder launch time, or via a file in the plugin root directory named `osumo_anonlogin.txt`, which should just contain one line of text with the username. Failing to supply these results in a fatal error, and the osumo plugin will not load.
- No password is required for the anonymous user account. Normal logins via the authentication subsystem are not possible. The way to become this user is to hit the `POST /osumo/anonlogin` endpoint.
- The plugin will create the anonymous user if it does not already exist.

### CODE REVIEW

**Comment by waxlamp** [comment]
@opadron this is ready for review.

File: web_client/actions.jsx [hunk 1]
```diff
@@ -529,14 +535,16 @@ export const submitLoginForm = (form) => promiseAction(
...
+        .then(() => rest.login(login, password))
         .then((user) => dispatch(setCurrentUser(user, user.token.token)))
         .then((user) => (result = user))
         .then(() => dispatch(closeDialog()))
         .then(() => result)
         .catch(
           ({ responseJSON: { message } }) => (
             dispatch(setDialogError('login', message))
+              .then(() => dispatch(loginAnonymousUser()))
           )
```
Comment (opadron) [inline web_client/actions.jsx]: It just occurred to me that we'd want to move the catch up a few calls in the chain.  We wouldn't want to try to anonymously log in if something went wrong with the dialog closing, for example.

**Review by opadron — CHANGES_REQUESTED** [review]
LGTM after that catch is moved up.

File: web_client/actions.jsx (hunk 1 above)
Comment (opadron) [reply web_client/actions.jsx]: I'll elaborate even further on this point:

What we want, *precisely*, is for a catch handler to handle the case where the normal `rest.login()` call fails.  In this case, the app needs to relog as the anonymous user.  The important detail is that we don't want this catch handler to fire in *any* other scenario.

To accomplish this, the promise handler that returns `rest.login(...)` (line 539) should be rewritten to return a promise subchain, with `rest.login(...)` at the head, and the catch handler being the only two links in the chain.

```javascript
return (
  rest
    .logout()
    .then(() => (
      rest.login(login, password)
        .catch(
          ({ responseJSON: { message } }) => (
            ...
          )
        )
    ))
    .then((user) => dispatch(...))
    ...
)
```

Please excuse any typos in the code above.  It has not been tested.

Comment (opadron) [reply web_client/actions.jsx]: I also just realized that the catch handler needs to return `Promise.reject(arg)` where arg is the argument passed to it.  This also needs to be done to propagate the error to the parent promise chain.

Comment (waxlamp) [reply web_client/actions.jsx]: @opadron could you take a look at b492118? Not sure I did it correctly.

**Review by opadron — APPROVED** [review]
LGTM

**Comment by jeffbaumes** [comment]
🍰 🎉

**Pull request merged**